files/boot-unseal.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/ash
+
+# Loop through all devices to find the ESP
+for device in $(blkid -o device); do
+    part_type=$(blkid -p $device -s PART_ENTRY_TYPE -o value)
+    if [[ "$part_type" != "c12a7328-f81f-11d2-ba4b-00a0c93ec93b" ]]; then
+        continue
+    fi
+
+    # Temporarily mount the ESP to read disk unlock keys
+    mkdir -p /mnt/esp
+    mount -t vfat $device /mnt/esp
+
+    # Try all of the keys in the appropriate folder
+    for f in /mnt/esp/*/disk_unlock_keys/*.sealed; do
+        if gotpm unseal --input "$f" --output "/crypto_keyfile.bin" ; then
+            echo "gotpm: Unsealed $f"
+        fi
```

Comment:
What if the first unsealed key cannot decrypt the disk? Will the system allow the script to try the next key in disk_unlock_keys?

Reply:
I added some additional to avoid unsealing multiple keys, and just succeeding when the first one is unsealed.

However, due to the structure of Arch Linux's initramfs hooks, we cannot check the disk unlock status without modifying the default `encrypt` hook shipped with the OS (like we do for Debian).

The `encrypt` hook supports an in-memory keyfile (defaulting to `/crypto_keyfile.bin`), but does not support _multiple_ keyfiles, so we just have to only try the first key and then fall back to password if that doesn't work.

It's not ideal, but it allows us to avoid reimplementing the main Arch Linux `encrypt` hook.